modules/quotas/main.tf

```diff
@@ -0,0 +1,45 @@
+##############################################################################
+# Upgrade container registry quota
+##############################################################################
+
+# Upgrade container registry storage quota
+resource "restapi_object" "container_registry_storage_quota" {
+  count = var.update_storage_quota ? 1 : 0
+  path  = "//${var.container_registry_endpoint}/api/v1/quotas"
+  data = jsonencode({
+    "storage_megabytes" : var.storage_megabytes
+  })
+  create_method  = "PATCH"
+  create_path    = "//${var.container_registry_endpoint}/api/v1/quotas"
+  destroy_method = "PATCH"
+  destroy_path   = "//${var.container_registry_endpoint}/api/v1/quotas"
+  destroy_data = jsonencode({
+    "storage_megabytes" : 500 # set to default 500 MB
```

Comment:
This might be ok but my worry here is that if a user does not start at 500mb the destroy is not reverting to the previous value. Account settings with IAM settings have a similar problem. We should check how that module handles it. @ocofaigh this is what I mentioned

Reply:
Here is what I was thinking of. The `ibm_iam_account_settings` resource supports setting the value `NOT_SET` for some account settings. And as per [provider docs](https://registry.terraform.io/providers/IBM-Cloud/ibm/1.55.0-beta0/docs/resources/iam_account_settings):

> NOT_SET - to 'unset' a previous set value.

I don't know how the provider is doing this? Are the storing the previous value in the state?

Reply:
We are using restapi provider here and there isn't any previous value in the state file, it's mostly the api response

Reply:
Created the provider issue -> https://github.com/IBM-Cloud/terraform-provider-ibm/issues/5816